app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,23 @@
+class AnnouncementsController < Admin::ApplicationController
+  before_action :set_announcement, only: [:show]
+
+  # GET /announcements
+  def index
+    @announcements = Announcement.all
+  end
+
+  def show
+
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_announcement
+      @announcement = Announcement.find(params[:id])
+    end
+
+    # Only allow a trusted parameter "white list" through.
+    def announcement_params
```

Comment:
since you don't need have update or create actions you don't need announcement_params method, you can remove it